app/utils/graphql.ts

```diff
@@ -153,31 +153,38 @@ export const useMutation = (mutation: RequestDocument) => {
 };
 
 /**
- * Hook which constructs a graphql query string to fetch data from a lookup table
+ * Hook which constructs a graphql query string to fetch data from a lookup table.
+ * 
+ * The query returned by this hook uses field name aliasing to ensure so that it
+ * always returns an array of objects with an `id` and `label` property, which 
+ * makes the results of this query compatible with our lookup value editing
+ * component.
  */
-export const useLookupQuery = (lookupTableDef: LookupTableDef | undefined) =>
+export const useLookupQuery = <T extends Record<string, unknown>>(
+  relationship?: Relationship<T>
+) =>
   useMemo(() => {
-    if (!lookupTableDef) {
+    if (!relationship) {
       return [];
     }
     // construct the Hasura typename, which is prefixed with the schema name
     // if schema is not public
     const prefix =
-      lookupTableDef.tableSchema === "public"
+      relationship.tableSchema === "public"
         ? ""
-        : lookupTableDef.tableSchema + "_";
+        : relationship.tableSchema + "_";
 
-    const typename = `${prefix}${lookupTableDef.tableName}`;
+    const typename = `${prefix}${relationship.tableName}`;
 
     return [
       gql`
         query LookupTableQuery {
-          ${typename}(order_by: {label: asc}) {
-            id
-            label
+          ${typename}(order_by: {${relationship.labelColumnName}: asc}) {
+            id: ${relationship.idColumnName}
+            label: ${relationship.labelColumnName}
```

Comment:
like i mention in the JSdoc for this hook—we are aliasing the `id` and `label` properties of this query builder, so that any lookup table we load will have the same shape. right now, we're only using lookups in the `lookups` schema in our db, which all of the same shape, but if we wanted to get lookup values to an EMS record, we could do that as well and still use the same UI form components.